### PR TITLE
Add filter support

### PR DESCRIPTION
**Pull request by jabuj**

### CODE REVIEW

File: examples/demo.js [hunk 1]
```diff
@@ -12,21 +12,24 @@ const format = workbook.addFormat();
...
-worksheet.writeNumber(2, 0, 123);
-worksheet.writeNumber(3, 0, 123.456);
+worksheet.writeNumber(1, 1, 123);
+worksheet.writeNumber(2, 1, 123.456);
+
+/* Change the column width for clarity. */
+worksheet.setColumn(1, 1, 20);
+
+/* Apply autofilters */
+worksheet.autofilter(0, 1, 2, 1);
```
Comment (mohd-akram) [inline examples/demo.js]: The examples are intended to produce the exact output as their C equivalent so we can compare them. You'll have to check out `examples/autofilter.c` and create a JS version. It looks like we might need an additional `worksheet.filterColumn` API for that (we can combine `worksheet_filter_column` and `worksheet_filter_column2` into one method with two overloads). I can do that, or you can add it to this PR.

Comment (jabuj) [reply examples/demo.js]: Would be nice if you could do it, writing for c, c++ and js simultaneously and doing my best not to leak memory is giving me brain damage little by little :) Although filter api seems simpler than data validation. I've removed autofilter from `demo.js` for now

Comment (mohd-akram) [reply examples/demo.js]: No worries. For these things it's best to run with the sanitizer while debugging, which makes it much easier to catch any memory issues. You can do this with `node-gyp rebuild --asan=1 --ubsan=1`.

**Comment by mohd-akram** [comment]
Thanks!

**Comment by mohd-akram** [comment]
Released in v0.2.4.

**Pull request merged**